lib/kafo/puppet_module.rb

```diff
@@ -12,7 +12,11 @@ class PuppetModule
     attr_reader :name, :identifier, :params, :dir_name, :class_name, :manifest_name, :manifest_path,
                 :groups, :params_path, :params_class_name, :configuration, :raw_data
 
-    def initialize(identifier, parser = KafoParsers::PuppetModuleParser, configuration = KafoConfigure.config)
+    def self.find_parser
+      @@parser ||= KafoParsers::Parsers.find_available(:logger => KafoConfigure.logger) || :none
```

Comment:
This is so it's cached when creating multiple PuppetModules, e.g. an installer with a dozen modules would call find_parser once for each module, which is very slow when using the puppet-strings parser (its available? method).  An instance variable wouldn't work across multiple instances of PuppetModule.

I could move this into Kafo::Configuration#modules instead?